flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -526,14 +553,21 @@ public void execute(@Nonnull Runnable command) {
         public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
             final long delayMillis = TimeUnit.MILLISECONDS.convert(delay, unit);
             FutureTask<Void> ft = new FutureTask<>(command, null);
-            if (mainScheduledExecutor.isShutdown()) {
-                log.warn(
-                        "The scheduled executor service is shutdown and ignores the command {}",
-                        command);
-            } else {
-                mainScheduledExecutor.schedule(
-                        () -> gateway.runAsync(ft), delayMillis, TimeUnit.MILLISECONDS);
-            }
+            getRunningFuture
```

Comment:
I think we should log a warning when the state is not RUNNING, rather than introducing `#getRunningFuture`, whose semantics don't seem very clear.